Add docs for project properties that control PDBs and XML publishing (#31473)
dotnet/docs: docs/core/project-sdk/msbuild-props.md

</PropertyGroup>
```

## Build output properties

- [CopyDebugSymbolFilesFromPackages](#copydebugsymbolfilesfrompackages)

> - [CopyDocumentationFilesFromPackages](#copydocumentationfilesfrompackages)

Reviewer: @baronfel What do you think about changing the existing "Compilation-related properties" to "Build-related properties" and adding these two properties there?

> - [CopyDocumentationFilesFromPackages](#copydocumentationfilesfrompackages)

Author: I was a little torn on that - those properties are all about the inputs sent to the compiler: files, compiler options, resource generation. These are odd in that they are post-compile, impacting only what ends up in the build output directory. So strictly speaking not _compile_ options, but I don't know a) how much a reader would be confused by that, and b) how much anyone other than me cares!

> - [CopyDocumentationFilesFromPackages](#copydocumentationfilesfrompackages)

Reviewer: I think my preference is to rename to "Build-related properties" and then move these two to that section instead of creating a new section.

> - [CopyDocumentationFilesFromPackages](#copydocumentationfilesfrompackages)

Author: Sounds good - I'll send those updates soon.

### CopyDebugSymbolFilesFromPackages

> When set to `true` this property causes all Symbol Files (also known as PDB files) from PackageReferences in the project to be copied to the build output. These can be used to provide more informative stack traces for exceptions, as well as making memory dumps and traces of the running application easier to understand, though including these files results in an increased deployment bundle size.

Reviewer: ```suggestion
When this property is set to `true`, all symbol files (also known as PDB files) from `PackageReference` items in the project are copied to the build output. These files can provide more informative stack traces for exceptions and make memory dumps and traces of the running application easier to understand. However, including these files results in an increased deployment bundle size.
```

This property was introduced in .NET SDK 7.0.100, though it defaults to not being specified.

### CopyDocumentationFilesFromPackages

> When set to `true` this property causes all generated XML Documentation from PackageReferences in the project to be copied to the build output. Note that enabling this feature will result in increased deployment bundle size.

Reviewer: ```suggestion
When this property is set to `true`, all generated XML documentation files from `PackageReference` items in the project are copied to the build output. Note that enabling this feature will result in increased deployment bundle size.
```

For more information, see [Write reference assemblies to intermediate output](../compatibility/sdk/6.0/write-reference-assemblies-to-obj.md).

### PublishDocumentationFile

> When this property is `true`, the XML Documentation file for the project (if one is created, see [GenerateDocumentationFile](#generatedocumentationfile)) will be included in the publish output for the project. This property defaults to `true`.

Reviewer: ```suggestion
When this property is `true`, the XML documentation file for the project, if one is generated, is included in the publish output for the project. This property defaults to `true`.

> [!TIP]
> Set [GenerateDocumentationFile](#generatedocumentationfile)) to `true` to generate an XML documentation file at compile time.
```

### PublishDocumentationFiles

> This property is an enablement flag for several other properties that control if various kinds of XML Documentation files are copied to the publish directory by default. See [PublishDocumentationFile](#publishdocumentationfile) and [PublishReferencesDocumentationFiles](#publishreferencesdocumentationfiles). If those properties are unset, and this property is set, then those properties will default to `true`. This property defaults to `true`.

Reviewer: ```suggestion
This property is an enablement flag for several other properties that control whether various kinds of XML documentation files are copied to the publish directory by default, namely [PublishDocumentationFile](#publishdocumentationfile) and [PublishReferencesDocumentationFiles](#publishreferencesdocumentationfiles). If those properties are unset, and this property is set, then those properties will default to `true`. This property defaults to `true`.
```

### PublishReferencesDocumentationFiles

> When this property is `true`, XML Documentation files for the project's references will be copied to the publish directory, not just runtime assets like DLL files.  This property defaults to `true`.

Reviewer: ```suggestion
When this property is `true`, XML documentation files for the project's references are copied to the publish directory, instead of just run-time assets like DLL files. This property defaults to `true`.
```
